hack/components/git-utils.sh

```diff
@@ -26,3 +26,8 @@ function git-utils::get_component_tag() {
         git describe --tags
     )
 }
+
+function git-utils::create_temp_path() {
+	TEMP_DIR=$(mktemp -d -p .)
+	trap "rm -rf ${TEMP_DIR}" EXIT
```

Comment:
Not sure if we can set a trap inside a function. Moreover, we cannot set multiple traps per EXIT. I'd say that the caller of this function has to ensure cleanup.

Or we can just drop this, it will be removed eventually.

Reply:
It works because I don't run it with a subshell. If it were to do the rm at the exit of the function - then the clone would have failed.

Reply:
Can you explain why you say there are multiple traps? I think each bump script runs standalone bt the MakeFile

Reply:
Yes, but you define the trap in utils. If the module importing this function and using it had predefined trap for EXIT, you would shadow it with this one.

Reply:
I can do the trap inside the father script, but I think we shouldn't because:
1. It works since I'm sourcing this script, so as long as I don't run it in a subshell, it should just run as if the lines are in the father script.
2. The point of using this function is to run the same lines in all the scripts, so moving the trap line will reduce maintainabilty

What do you think? 😀

Reply:
> Yes, but you define the trap in utils. If the module importing this function and using it had predefined trap for EXIT, you would shadow it with this one.

Oh, I get it.. You mean this way it's hidden.. OK agree with that